src/main/java/com/xebia/incubator/xebium/SeleniumDriverFixture.java

```diff
@@ -413,19 +425,75 @@ private String unalias(String value) {
 			String alias = value.substring(ALIAS_PREFIX.length());
 			String subst = aliases.get(alias);
 			if (subst != null) {
-			    LOG.info("Expanded alias '" + alias + "' to '" + result + "'");
-			    result = subst;
+				LOG.info("Expanded alias '" + alias + "' to '" + result + "'");
+				result = subst;
 			}
 		}
 		return result;
 	}
 
+	// The Below 2 functions (searchAllFrames) are used to search all frames for a specific xpath search, and also switchTo the frame that has the found web element
+
+	public WebElement searchAllFrames(String elementXpath) {
+		return searchAllFrames(getWebDriver(), elementXpath, new ArrayList<WebElement>());
+	}
+
+	public WebElement searchAllFrames(WebDriver driver, String elementXpath, List <WebElement> frameSwitches) {
+		if (frameSwitches.size() == 0) //if starting then make sure to search from top
+			driver.switchTo().defaultContent();
+		WebElement element = null; //return null if not found or error during search.
+		try {
+			element = driver.findElement(By.xpath(elementXpath));// need to catch in case not found
+		}catch(Exception e) {}
+		if (element == null)
+		{
+			List <WebElement> framesList = null;
+			try {
+				framesList = driver.findElements(By.xpath("//iframe|//frame"));
+				framesList = driver.findElements(By.cssSelector("iframe|frame"));
+			}catch(Exception e) {}
+			if (framesList == null)
+				return null;
+			for(WebElement frame:framesList){
+				// need to start from the top and switch back to next frame
+				driver.switchTo().defaultContent();
+				for(WebElement nestedFrame:frameSwitches){//nested frames list
+					driver.switchTo().frame(nestedFrame);
+				}
+				driver.switchTo().frame(frame);
+
+				frameSwitches.add(frame);
+				element =  searchAllFrames(driver, elementXpath, frameSwitches);
+				if (element != null)
+					return element; // element found
+				else
+					frameSwitches.remove(frame); //this frame had nothing so remove it from the list
+			}
+			return null; //element not found
+		}
+		else
+			return element; //element found
+	}
+
 	private boolean executeDoCommand(final String methodName, final String[] values) {
 
+		String output = null;
+		boolean result = true;
+
 		ExtendedSeleniumCommand command = new ExtendedSeleniumCommand(methodName);
 
+		boolean locatorFlag = locatorCheck.verifyElementPresent(command, values);
+
 		SeleniumCommandResult commandResult;
 
+		if (!locatorFlag) { // if it is a locator command and verifyElementPresent fails
```

Comment:
I don't quite understand what's supposed to happen here: 'output' and 'result' never seem to be used, and you seem to put a lot of effort into determining this 'locatorFlag' (including recursively searching through frames) - only to then ignore that value?